[{"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/8164", "id": 1608597672, "node_id": "I_kwDOA8iScc5f4UCo", "number": 8164, "title": "Amazon Requests", "user": {"login": "thomasthaddeus", "id": 92204097, "node_id": "U_kgDOBX7sQQ", "avatar_url": "https://avatars.githubusercontent.com/u/92204097?v=4", "gravatar_id": "", "url": "https://api.github.com/users/thomasthaddeus", "html_url": "https://github.com/thomasthaddeus", "followers_url": "https://api.github.com/users/thomasthaddeus/followers", "following_url": "https://api.github.com/users/thomasthaddeus/following{/other_user}", "gists_url": "https://api.github.com/users/thomasthaddeus/gists{/gist_id}", "starred_url": "https://api.github.com/users/thomasthaddeus/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/thomasthaddeus/subscriptions", "organizations_url": "https://api.github.com/users/thomasthaddeus/orgs", "repos_url": "https://api.github.com/users/thomasthaddeus/repos", "events_url": "https://api.github.com/users/thomasthaddeus/events{/privacy}", "received_events_url": "https://api.github.com/users/thomasthaddeus/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 8, "created_at": "2023-03-03T13:05:36Z", "updated_at": "2023-04-07T11:41:57Z", "closed_at": "2023-03-14T11:37:47Z", "author_association": "NONE", "active_lock_reason": null, "body": "### Repository commit\n\n222715391-2b3d3a33-d553-481c-8577-56950e913eaf\n\n### Python version (python --version)\n\npython 3.9\n\n### Dependencies version (pip freeze)\n\npython=3.9\r\nrequests not in loader\n\n### Expected behavior\n\nI'm not exactly sure other than I loaded it on the website, and this was the default behavior.\n\n### Actual behavior\n\n[![image](https://user-images.githubusercontent.com/92204097/222715391-2b3d3a33-d553-481c-8577-56950e913eaf.png)](https://the-algorithms.com/playground?id=6869)\r\n", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/8164/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/7987", "id": 1449855808, "node_id": "I_kwDOA8iScc5WawtA", "number": 7987, "title": "Fixes required for new version of mypy #7983", "user": {"login": "cclauss", "id": 3709715, "node_id": "MDQ6VXNlcjM3MDk3MTU=", "avatar_url": "https://avatars.githubusercontent.com/u/3709715?v=4", "gravatar_id": "", "url": "https://api.github.com/users/cclauss", "html_url": "https://github.com/cclauss", "followers_url": "https://api.github.com/users/cclauss/followers", "following_url": "https://api.github.com/users/cclauss/following{/other_user}", "gists_url": "https://api.github.com/users/cclauss/gists{/gist_id}", "starred_url": "https://api.github.com/users/cclauss/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/cclauss/subscriptions", "organizations_url": "https://api.github.com/users/cclauss/orgs", "repos_url": "https://api.github.com/users/cclauss/repos", "events_url": "https://api.github.com/users/cclauss/events{/privacy}", "received_events_url": "https://api.github.com/users/cclauss/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": {"login": "dhruvmanila", "id": 67177269, "node_id": "MDQ6VXNlcjY3MTc3MjY5", "avatar_url": "https://avatars.githubusercontent.com/u/67177269?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dhruvmanila", "html_url": "https://github.com/dhruvmanila", "followers_url": "https://api.github.com/users/dhruvmanila/followers", "following_url": "https://api.github.com/users/dhruvmanila/following{/other_user}", "gists_url": "https://api.github.com/users/dhruvmanila/gists{/gist_id}", "starred_url": "https://api.github.com/users/dhruvmanila/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dhruvmanila/subscriptions", "organizations_url": "https://api.github.com/users/dhruvmanila/orgs", "repos_url": "https://api.github.com/users/dhruvmanila/repos", "events_url": "https://api.github.com/users/dhruvmanila/events{/privacy}", "received_events_url": "https://api.github.com/users/dhruvmanila/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "dhruvmanila", "id": 67177269, "node_id": "MDQ6VXNlcjY3MTc3MjY5", "avatar_url": "https://avatars.githubusercontent.com/u/67177269?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dhruvmanila", "html_url": "https://github.com/dhruvmanila", "followers_url": "https://api.github.com/users/dhruvmanila/followers", "following_url": "https://api.github.com/users/dhruvmanila/following{/other_user}", "gists_url": "https://api.github.com/users/dhruvmanila/gists{/gist_id}", "starred_url": "https://api.github.com/users/dhruvmanila/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dhruvmanila/subscriptions", "organizations_url": "https://api.github.com/users/dhruvmanila/orgs", "repos_url": "https://api.github.com/users/dhruvmanila/repos", "events_url": "https://api.github.com/users/dhruvmanila/events{/privacy}", "received_events_url": "https://api.github.com/users/dhruvmanila/received_events", "type": "User", "site_admin": false}], "milestone": null, "comments": 0, "created_at": "2022-11-15T14:14:10Z", "updated_at": "2022-11-15T17:29:15Z", "closed_at": "2022-11-15T17:29:15Z", "author_association": "MEMBER", "active_lock_reason": null, "body": "### Repository commit\r\n\r\nmaster\r\n\r\n### Python version (python --version)\r\n\r\nPython 3.11.0\r\n\r\n### Dependencies version (pip freeze)\r\n\r\nYes\r\n\r\n### Expected behavior\r\n\r\nmypy passes.\r\n\r\n### Actual behavior\r\n\r\n# Issue #7983\r\n\r\n@CaedenPH", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7987/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/7942", "id": 1432689224, "node_id": "I_kwDOA8iScc5VZRpI", "number": 7942, "title": "linear_algebra/src/polynom_for_points.py is too complex", "user": {"login": "cclauss", "id": 3709715, "node_id": "MDQ6VXNlcjM3MDk3MTU=", "avatar_url": "https://avatars.githubusercontent.com/u/3709715?v=4", "gravatar_id": "", "url": "https://api.github.com/users/cclauss", "html_url": "https://github.com/cclauss", "followers_url": "https://api.github.com/users/cclauss/followers", "following_url": "https://api.github.com/users/cclauss/following{/other_user}", "gists_url": "https://api.github.com/users/cclauss/gists{/gist_id}", "starred_url": "https://api.github.com/users/cclauss/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/cclauss/subscriptions", "organizations_url": "https://api.github.com/users/cclauss/orgs", "repos_url": "https://api.github.com/users/cclauss/repos", "events_url": "https://api.github.com/users/cclauss/events{/privacy}", "received_events_url": "https://api.github.com/users/cclauss/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2022-11-02T08:14:57Z", "updated_at": "2022-11-02T18:40:26Z", "closed_at": "2022-11-02T18:40:26Z", "author_association": "MEMBER", "active_lock_reason": null, "body": "See: https://pypi.org/project/mccabe\r\n\r\n### Repository commit\r\n\r\nmaster\r\n\r\n### Python version (python --version)\r\n\r\nPython 3.11.0\r\n\r\n### Dependencies version (pip freeze)\r\n\r\nNone\r\n\r\n### Expected behavior\r\n\r\n% `flake8 --select=C --max-complexity=21 .`  # --> no errors.\r\n\r\n### Actual behavior\r\n\r\n`linear_algebra/src/polynom_for_points.py:1:1: C901 'points_to_polynomial' is too complex (23)`\r\n\r\nReduce the complexity of `points_to_polynomial()` and then lower the `--max-complexity` threshold in the file `.flake8`.", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7942/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/7512", "id": 1419243875, "node_id": "I_kwDOA8iScc5Ul_Fj", "number": 7512, "title": "The git pre-commit hook contradices itself and goes beyond the scope of the files to commit", "user": {"login": "carlos3dx", "id": 18118830, "node_id": "MDQ6VXNlcjE4MTE4ODMw", "avatar_url": "https://avatars.githubusercontent.com/u/18118830?v=4", "gravatar_id": "", "url": "https://api.github.com/users/carlos3dx", "html_url": "https://github.com/carlos3dx", "followers_url": "https://api.github.com/users/carlos3dx/followers", "following_url": "https://api.github.com/users/carlos3dx/following{/other_user}", "gists_url": "https://api.github.com/users/carlos3dx/gists{/gist_id}", "starred_url": "https://api.github.com/users/carlos3dx/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/carlos3dx/subscriptions", "organizations_url": "https://api.github.com/users/carlos3dx/orgs", "repos_url": "https://api.github.com/users/carlos3dx/repos", "events_url": "https://api.github.com/users/carlos3dx/events{/privacy}", "received_events_url": "https://api.github.com/users/carlos3dx/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 10, "created_at": "2022-10-22T11:11:44Z", "updated_at": "2022-10-29T21:20:07Z", "closed_at": "2022-10-29T21:20:07Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "### Repository commit\r\n\r\ne6d825eaee9d3ad86e30c415a1f7da36ed39351c\r\n\r\n### Python version (python --version)\r\n\r\nPython 3.9.1\r\n\r\n### Dependencies version (pip freeze)\r\n\r\nabsl-py==1.3.0\r\nastunparse==1.6.3\r\nbeautifulsoup4==4.11.1\r\ncachetools==5.2.0\r\ncertifi==2022.9.24\r\ncffi==1.15.1\r\ncharset-normalizer==2.1.1\r\ncontourpy==1.0.5\r\ncryptography==38.0.1\r\ncycler==0.11.0\r\ndill==0.3.5.1\r\nfake-useragent==0.1.11\r\nflatbuffers==22.9.24\r\nfonttools==4.37.4\r\ngast==0.4.0\r\ngoogle-auth==2.13.0\r\ngoogle-auth-oauthlib==0.4.6\r\ngoogle-pasta==0.2.0\r\ngrpcio==1.49.1\r\nh5py==3.7.0\r\nidna==3.4\r\nimportlib-metadata==5.0.0\r\njoblib==1.2.0\r\nkeras==2.10.0\r\nKeras-Preprocessing==1.1.2\r\nkiwisolver==1.4.4\r\nlibclang==14.0.6\r\nlxml==4.9.1\r\nMarkdown==3.4.1\r\nMarkupSafe==2.1.1\r\nmatplotlib==3.6.1\r\nmpmath==1.2.1\r\nnetworkx==2.8.7\r\nntlm-auth==1.5.0\r\nnumpy==1.23.4\r\noauthlib==3.2.1\r\nopencv-python==4.6.0.66\r\nopt-einsum==3.3.0\r\npackaging==21.3\r\npandas==1.5.0\r\npatsy==0.5.3\r\npbr==5.10.0\r\nPillow==9.2.0\r\npip==21.3.1\r\nply==3.11\r\nprojectq==0.8.0\r\nprotobuf==3.19.6\r\npsutil==5.9.2\r\npyasn1==0.4.8\r\npyasn1-modules==0.2.8\r\npycparser==2.21\r\npyparsing==3.0.9\r\npython-dateutil==2.8.2\r\npytz==2022.4\r\nqiskit==0.39.0\r\nqiskit-aer==0.11.0\r\nqiskit-ibmq-provider==0.19.2\r\nqiskit-terra==0.22.0\r\nrequests==2.28.1\r\nrequests-ntlm==1.1.0\r\nrequests-oauthlib==1.3.1\r\nretworkx==0.12.0\r\nrsa==4.9\r\nrustworkx==0.12.0\r\nscikit-fuzzy==0.4.2\r\nscikit-learn==1.1.2\r\nscipy==1.9.2\r\nsetuptools==60.2.0\r\nsix==1.16.0\r\nsklearn==0.0\r\nsoupsieve==2.3.2.post1\r\nstatsmodels==0.13.2\r\nstevedore==4.0.1\r\nsymengine==0.9.2\r\nsympy==1.11.1\r\ntensorboard==2.10.1\r\ntensorboard-data-server==0.6.1\r\ntensorboard-plugin-wit==1.8.1\r\ntensorflow==2.10.0\r\ntensorflow-estimator==2.10.0\r\ntensorflow-io-gcs-filesystem==0.27.0\r\ntermcolor==2.0.1\r\ntexttable==1.6.4\r\nthreadpoolctl==3.1.0\r\ntk==0.1.0\r\ntweedledum==1.1.1\r\ntweepy==4.10.1\r\ntyping_extensions==4.4.0\r\nurllib3==1.26.12\r\nwebsocket-client==1.4.1\r\nwebsockets==10.3\r\nWerkzeug==2.2.2\r\nwheel==0.37.1\r\nwrapt==1.14.1\r\nxgboost==1.6.2\r\nyulewalker==0.1.1\r\nzipp==3.9.0\r\n\r\n\r\n### Expected behavior\r\n\r\nWhen executing the pre-commit hook it only should only focus on the files that are being committed and ignore the ones in fit ignored folder like env.\r\n\r\nType hinting with typing library should be respected.\r\n\r\n### Actual behavior\r\n\r\nWhen executing the pre-commit hook the file formatter complains about files that are not going to be committed and are in an ignored folder, for example, env.\r\n<img width=\"757\" alt=\"Captura de Pantalla 2022-10-22 a las 14 06 44\" src=\"https://user-images.githubusercontent.com/18118830/197338220-fae5f2af-6e58-4ffe-98f4-ceee02739d3b.png\">\r\n\r\nAlso if a function has type hinting from typing, it converts `foo:List[str]` to `foo:list[str]`, and later complains that it is not subscriptable and also for the the unused import `from typing import List` cause the type hinting is not longer using the imported one. But if you don't do type hinting, the suggested hint includes the ones from typing", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/7512/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/6916", "id": 1403162948, "node_id": "I_kwDOA8iScc5TopFE", "number": 6916, "title": "`maths\\jaccard_similarity.py` does not validate input and can divide by zero", "user": {"login": "CenTdemeern1", "id": 24682280, "node_id": "MDQ6VXNlcjI0NjgyMjgw", "avatar_url": "https://avatars.githubusercontent.com/u/24682280?v=4", "gravatar_id": "", "url": "https://api.github.com/users/CenTdemeern1", "html_url": "https://github.com/CenTdemeern1", "followers_url": "https://api.github.com/users/CenTdemeern1/followers", "following_url": "https://api.github.com/users/CenTdemeern1/following{/other_user}", "gists_url": "https://api.github.com/users/CenTdemeern1/gists{/gist_id}", "starred_url": "https://api.github.com/users/CenTdemeern1/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/CenTdemeern1/subscriptions", "organizations_url": "https://api.github.com/users/CenTdemeern1/orgs", "repos_url": "https://api.github.com/users/CenTdemeern1/repos", "events_url": "https://api.github.com/users/CenTdemeern1/events{/privacy}", "received_events_url": "https://api.github.com/users/CenTdemeern1/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 5, "created_at": "2022-10-10T13:16:05Z", "updated_at": "2022-10-29T21:20:28Z", "closed_at": "2022-10-29T21:20:28Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "### Repository commit\n\n51dba4d743cd2c8d407eea3e9cd4e7b2f69ee34d\n\n### Python version (python --version)\n\nPython 3.10.2\n\n### Dependencies version (pip freeze)\n\npip==22.2.2\r\nsetuptools==60.10.0\r\nwheel==0.37.1\n\n### Expected behavior\n\nAccording to `CONTRIBUTING.md`, a `ValueError` should be raised in the case of erroneous input.\r\n![image](https://user-images.githubusercontent.com/24682280/194873800-0f430e1c-972c-4b56-b41c-38bd81408379.png)\n\n### Actual behavior\n\nThis code makes no effort to check the input, and this can result in a zero division when given empty lists as input:\r\n![image](https://user-images.githubusercontent.com/24682280/194874199-7d8c5153-5fa0-4f7b-a3b9-13e2538a527e.png)", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/6916/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/4764", "id": 996587206, "node_id": "I_kwDOA8iScc47ZrbG", "number": 4764, "title": "Bug with union in disjoint_set", "user": {"login": "yunhenk", "id": 11896932, "node_id": "MDQ6VXNlcjExODk2OTMy", "avatar_url": "https://avatars.githubusercontent.com/u/11896932?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yunhenk", "html_url": "https://github.com/yunhenk", "followers_url": "https://api.github.com/users/yunhenk/followers", "following_url": "https://api.github.com/users/yunhenk/following{/other_user}", "gists_url": "https://api.github.com/users/yunhenk/gists{/gist_id}", "starred_url": "https://api.github.com/users/yunhenk/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yunhenk/subscriptions", "organizations_url": "https://api.github.com/users/yunhenk/orgs", "repos_url": "https://api.github.com/users/yunhenk/repos", "events_url": "https://api.github.com/users/yunhenk/events{/privacy}", "received_events_url": "https://api.github.com/users/yunhenk/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 902564907, "node_id": "MDU6TGFiZWw5MDI1NjQ5MDc=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/good%20first%20issue", "name": "good first issue", "color": "50d147", "default": true, "description": ""}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2021-09-15T02:14:12Z", "updated_at": "2021-09-24T10:55:23Z", "closed_at": "2021-09-24T10:55:23Z", "author_association": "NONE", "active_lock_reason": null, "body": "https://github.com/TheAlgorithms/Python/blob/master/data_structures/disjoint_set/disjoint_set.py\r\n```python\r\ndef union_set(x, y):\r\n    \"\"\"\r\n    union two sets.\r\n    set with bigger rank should be parent, so that the\r\n    disjoint set tree will be more flat.\r\n    \"\"\"\r\n    x, y = find_set(x), find_set(y)\r\n    if x.rank > y.rank:\r\n        y.parent = x\r\n    else:\r\n        x.parent = y\r\n        if x.rank == y.rank:\r\n            y.rank += 1\r\n```\r\n\r\nhere need check if x==y\r\n", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764/reactions", "total_count": 3, "+1": 3, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/4764/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/3944", "id": 749385984, "node_id": "MDU6SXNzdWU3NDkzODU5ODQ=", "number": 3944, "title": "web_programming/fetch_github_info.py err", "user": {"login": "Ksengine", "id": 63494497, "node_id": "MDQ6VXNlcjYzNDk0NDk3", "avatar_url": "https://avatars.githubusercontent.com/u/63494497?v=4", "gravatar_id": "", "url": "https://api.github.com/users/Ksengine", "html_url": "https://github.com/Ksengine", "followers_url": "https://api.github.com/users/Ksengine/followers", "following_url": "https://api.github.com/users/Ksengine/following{/other_user}", "gists_url": "https://api.github.com/users/Ksengine/gists{/gist_id}", "starred_url": "https://api.github.com/users/Ksengine/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/Ksengine/subscriptions", "organizations_url": "https://api.github.com/users/Ksengine/orgs", "repos_url": "https://api.github.com/users/Ksengine/repos", "events_url": "https://api.github.com/users/Ksengine/events{/privacy}", "received_events_url": "https://api.github.com/users/Ksengine/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-11-24T06:30:26Z", "updated_at": "2020-11-25T06:36:16Z", "closed_at": "2020-11-25T05:13:15Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "<pre>message: Bad credentials. The API can&apos;t be accessed using username/password authentication. Please create a personal access token to access this endpoint: http://github.com/settings/tokens\r\ndocumentation_url: https://docs.github.com/articles/creating-a-personal-access-token-for-the-command-line\r\n</pre>", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3944/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/3163", "id": 718664873, "node_id": "MDU6SXNzdWU3MTg2NjQ4NzM=", "number": 3163, "title": "Fix needed for Project Euler problem 234 sol1.py", "user": {"login": "dhruvmanila", "id": 67177269, "node_id": "MDQ6VXNlcjY3MTc3MjY5", "avatar_url": "https://avatars.githubusercontent.com/u/67177269?v=4", "gravatar_id": "", "url": "https://api.github.com/users/dhruvmanila", "html_url": "https://github.com/dhruvmanila", "followers_url": "https://api.github.com/users/dhruvmanila/followers", "following_url": "https://api.github.com/users/dhruvmanila/following{/other_user}", "gists_url": "https://api.github.com/users/dhruvmanila/gists{/gist_id}", "starred_url": "https://api.github.com/users/dhruvmanila/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/dhruvmanila/subscriptions", "organizations_url": "https://api.github.com/users/dhruvmanila/orgs", "repos_url": "https://api.github.com/users/dhruvmanila/repos", "events_url": "https://api.github.com/users/dhruvmanila/events{/privacy}", "received_events_url": "https://api.github.com/users/dhruvmanila/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151094, "node_id": "MDU6TGFiZWw0MDkxNTEwOTQ=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/help%20wanted", "name": "help wanted", "color": "128A0C", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 2, "created_at": "2020-10-10T17:23:25Z", "updated_at": "2020-10-11T17:46:17Z", "closed_at": "2020-10-11T17:46:17Z", "author_association": "MEMBER", "active_lock_reason": null, "body": "### Problem:\r\n\r\nAccording to the [problem description](https://projecteuler.net/problem=234), the input value for the function `solution()` should be `999966663333`. Now, if we follow the control flow of the problem we see that there's going to be a loop that will have `999966663333` many iterations and for each iteration [this line](https://github.com/TheAlgorithms/Python/blob/master/project_euler/problem_234/sol1.py#L43) asks the user for input. As mentioned in CONTRIBUTING.md, we cannot ask for input and this solution will be asking for the input for `999966663333` times.\r\n\r\n```python\r\nfor x in range(n):  # n = 999966663333 according to the problem description\r\n    l = [i for i in input().split()]  # noqa: E741\r\n```\r\n\r\n### Proposed solution:\r\n\r\n1. Remove the current solution and implement a new one.\r\n2. Take a look at the current solution and fix it.\r\n\r\nThe second solution is highly unlikely as I am not able to deduce what is actually going on in the function and why do we even need to ask for the input. If anyone can figure it out, they can submit a PR for this.\r\n\r\n***I propose the first solution.***\r\n\r\nOnce the issue is fixed, we can start running this [script](https://github.com/TheAlgorithms/Python/blob/master/project_euler/validate_solutions.py) which checks for all answers to Project Euler problems.", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/3163/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/2257", "id": 670617393, "node_id": "MDU6SXNzdWU2NzA2MTczOTM=", "number": 2257, "title": "arithmetic_analysis/lu_decomposition is wrong", "user": {"login": "spamegg1", "id": 4255997, "node_id": "MDQ6VXNlcjQyNTU5OTc=", "avatar_url": "https://avatars.githubusercontent.com/u/4255997?v=4", "gravatar_id": "", "url": "https://api.github.com/users/spamegg1", "html_url": "https://github.com/spamegg1", "followers_url": "https://api.github.com/users/spamegg1/followers", "following_url": "https://api.github.com/users/spamegg1/following{/other_user}", "gists_url": "https://api.github.com/users/spamegg1/gists{/gist_id}", "starred_url": "https://api.github.com/users/spamegg1/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/spamegg1/subscriptions", "organizations_url": "https://api.github.com/users/spamegg1/orgs", "repos_url": "https://api.github.com/users/spamegg1/repos", "events_url": "https://api.github.com/users/spamegg1/events{/privacy}", "received_events_url": "https://api.github.com/users/spamegg1/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151094, "node_id": "MDU6TGFiZWw0MDkxNTEwOTQ=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/help%20wanted", "name": "help wanted", "color": "128A0C", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2020-08-01T07:29:53Z", "updated_at": "2020-08-01T20:11:40Z", "closed_at": "2020-08-01T20:11:40Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "The provided example matrix in the file is\r\n```\r\n[[2, -2, 1], \r\n[0, 1, 2], \r\n[5, 3, 1]]\r\n```\r\nThe algorithm returns\r\n```\r\nL = [[1.  0.  0. ]\r\n [0.  1.  0. ]\r\n [2.5 0.  1. ]]\r\nU = [[ 2.  -2.   1. ]\r\n [ 0.   1.   2. ]\r\n [ 0.   8.  -1.5]]\r\n```\r\nMultiplying `L` and `U` does give us back the original matrix. \r\n\r\nHowever as you can see the matrix `U` is NOT upper-triangular. \r\n\r\n[Wikipedia](https://en.wikipedia.org/wiki/LU_decomposition) says that a matrix admits LU-decomposition if and only if all its principal minors (in this case, the determinants of all the 2x2 matrices obtained by removing a row and a column) are nonzero. Upon inspection we can see that this holds true of the original matrix, so it should definitely admit LU-decomposition.\r\n\r\nAn [online calculator gives](https://matrixcalc.org/en/#LU-decomposition%28%7B%7B2,-2,1%7D,%7B0,1,2%7D,%7B5,3,1%7D%7D%29)\r\n```\r\nL = [[1.  0.  0. ]\r\n       [0.  1.  0. ]\r\n       [2.5 8.  1. ]]\r\nU = [[ 2.  -2.   1. ]\r\n       [ 0.   1.   2. ]\r\n       [ 0.   0.  -17.5]]\r\n```\r\nJust to be sure I tried it on another matrix (it satisfies the principal minors condition):\r\n```\r\n[[1, 2, 3], \r\n[4, 5, 6], \r\n[7, 8, 9]]\r\n```\r\nThe algorithm gives\r\n```\r\nL = [[1.  0.  0. ]\r\n       [0.  1.  0. ]\r\n       [7. 0.  1. ]]\r\nU = [[ 1.  2.   3. ]\r\n       [ 4.   5.   6. ]\r\n       [ 0.   -6.  -12.]]\r\n```\r\nOnce again `U` is not upper-triangular.\r\n\r\nThe [online calculator gives](https://matrixcalc.org/en/#LU-decomposition%28%7B%7B1,2,3%7D,%7B4,5,6%7D,%7B7,8,9%7D%7D%29)\r\n```\r\nL = [[1.  0.  0. ]\r\n       [4.  1.  0. ]\r\n       [7. 2.  1. ]]\r\nU = [[ 1.  2.   3. ]\r\n       [ 0.   -3.   -6. ]\r\n       [ 0.   0.  0.]]\r\n```\r\n\r\nI'm not sure how to fix this algorithm, so help is definitely wanted.", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2257/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/2050", "id": 627797835, "node_id": "MDU6SXNzdWU2Mjc3OTc4MzU=", "number": 2050, "title": "Formatting issues not getting caught in TravisCI", "user": {"login": "onlinejudge95", "id": 44158581, "node_id": "MDQ6VXNlcjQ0MTU4NTgx", "avatar_url": "https://avatars.githubusercontent.com/u/44158581?v=4", "gravatar_id": "", "url": "https://api.github.com/users/onlinejudge95", "html_url": "https://github.com/onlinejudge95", "followers_url": "https://api.github.com/users/onlinejudge95/followers", "following_url": "https://api.github.com/users/onlinejudge95/following{/other_user}", "gists_url": "https://api.github.com/users/onlinejudge95/gists{/gist_id}", "starred_url": "https://api.github.com/users/onlinejudge95/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/onlinejudge95/subscriptions", "organizations_url": "https://api.github.com/users/onlinejudge95/orgs", "repos_url": "https://api.github.com/users/onlinejudge95/repos", "events_url": "https://api.github.com/users/onlinejudge95/events{/privacy}", "received_events_url": "https://api.github.com/users/onlinejudge95/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2020-05-30T19:03:30Z", "updated_at": "2020-05-31T04:51:10Z", "closed_at": "2020-05-31T04:51:10Z", "author_association": "COLLABORATOR", "active_lock_reason": null, "body": "It has come into the observation that files that are not `black` compliant are still passing the Travis CI build. Due to which our master has failing checks. Can we do something about it?\r\n\r\nReference https://github.com/TheAlgorithms/Python/pull/2041", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/2050/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745/events", "html_url": "https://github.com/TheAlgorithms/Python/pull/1745", "id": 562746529, "node_id": "MDExOlB1bGxSZXF1ZXN0MzczMjg2ODcx", "number": 1745, "title": "Fixes LGTM issues", "user": {"login": "onlinejudge95", "id": 44158581, "node_id": "MDQ6VXNlcjQ0MTU4NTgx", "avatar_url": "https://avatars.githubusercontent.com/u/44158581?v=4", "gravatar_id": "", "url": "https://api.github.com/users/onlinejudge95", "html_url": "https://github.com/onlinejudge95", "followers_url": "https://api.github.com/users/onlinejudge95/followers", "following_url": "https://api.github.com/users/onlinejudge95/following{/other_user}", "gists_url": "https://api.github.com/users/onlinejudge95/gists{/gist_id}", "starred_url": "https://api.github.com/users/onlinejudge95/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/onlinejudge95/subscriptions", "organizations_url": "https://api.github.com/users/onlinejudge95/orgs", "repos_url": "https://api.github.com/users/onlinejudge95/repos", "events_url": "https://api.github.com/users/onlinejudge95/events{/privacy}", "received_events_url": "https://api.github.com/users/onlinejudge95/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151093, "node_id": "MDU6TGFiZWw0MDkxNTEwOTM=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/enhancement", "name": "enhancement", "color": "696969", "default": true, "description": "This PR modified some existing files."}, {"id": 683905342, "node_id": "MDU6TGFiZWw2ODM5MDUzNDI=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/awaiting%20reviews", "name": "awaiting reviews", "color": "f9d0c4", "default": false, "description": "This PR is ready to be reviewed."}], "state": "closed", "locked": false, "assignee": {"login": "onlinejudge95", "id": 44158581, "node_id": "MDQ6VXNlcjQ0MTU4NTgx", "avatar_url": "https://avatars.githubusercontent.com/u/44158581?v=4", "gravatar_id": "", "url": "https://api.github.com/users/onlinejudge95", "html_url": "https://github.com/onlinejudge95", "followers_url": "https://api.github.com/users/onlinejudge95/followers", "following_url": "https://api.github.com/users/onlinejudge95/following{/other_user}", "gists_url": "https://api.github.com/users/onlinejudge95/gists{/gist_id}", "starred_url": "https://api.github.com/users/onlinejudge95/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/onlinejudge95/subscriptions", "organizations_url": "https://api.github.com/users/onlinejudge95/orgs", "repos_url": "https://api.github.com/users/onlinejudge95/repos", "events_url": "https://api.github.com/users/onlinejudge95/events{/privacy}", "received_events_url": "https://api.github.com/users/onlinejudge95/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "onlinejudge95", "id": 44158581, "node_id": "MDQ6VXNlcjQ0MTU4NTgx", "avatar_url": "https://avatars.githubusercontent.com/u/44158581?v=4", "gravatar_id": "", "url": "https://api.github.com/users/onlinejudge95", "html_url": "https://github.com/onlinejudge95", "followers_url": "https://api.github.com/users/onlinejudge95/followers", "following_url": "https://api.github.com/users/onlinejudge95/following{/other_user}", "gists_url": "https://api.github.com/users/onlinejudge95/gists{/gist_id}", "starred_url": "https://api.github.com/users/onlinejudge95/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/onlinejudge95/subscriptions", "organizations_url": "https://api.github.com/users/onlinejudge95/orgs", "repos_url": "https://api.github.com/users/onlinejudge95/repos", "events_url": "https://api.github.com/users/onlinejudge95/events{/privacy}", "received_events_url": "https://api.github.com/users/onlinejudge95/received_events", "type": "User", "site_admin": false}], "milestone": null, "comments": 2, "created_at": "2020-02-10T18:30:09Z", "updated_at": "2020-02-11T08:43:38Z", "closed_at": "2020-02-10T21:23:20Z", "author_association": "COLLABORATOR", "active_lock_reason": null, "draft": false, "pull_request": {"url": "https://api.github.com/repos/TheAlgorithms/Python/pulls/1745", "html_url": "https://github.com/TheAlgorithms/Python/pull/1745", "diff_url": "https://github.com/TheAlgorithms/Python/pull/1745.diff", "patch_url": "https://github.com/TheAlgorithms/Python/pull/1745.patch", "merged_at": "2020-02-10T21:23:20Z"}, "body": "### **Describe your change:**\r\nFixes following LGTM issues\r\n1. [Implementing __eq__](https://lgtm.com/projects/g/TheAlgorithms/Python/snapshot/ee5e954b9ed82bbe885bf42bbc446c213a43bd4e/files/searches/hill_climbing.py#x2978473143457f06:1)\r\n2. [Multiple definitions](https://lgtm.com/projects/g/TheAlgorithms/Python/snapshot/ee5e954b9ed82bbe885bf42bbc446c213a43bd4e/files/dynamic_programming/max_sum_contiguous_subsequence.py?sort=name&dir=ASC&mode=heatmap#x5e72d15036130bb6:1)\r\n\r\n* [ ] Add an algorithm?\r\n* [x] Fix a bug or typo in an existing algorithm?\r\n* [ ] Documentation change?\r\n\r\n### **Checklist:**\r\n* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).\r\n* [ ] This pull request is all my own work -- I have not plagiarized.\r\n* [x] I know that pull requests will not be merged if they fail the automated tests.\r\n* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.\r\n* [ ] All new Python files are placed inside an existing directory.\r\n* [ ] All filenames are in all lowercase characters with no spaces or dashes.\r\n* [ ] All functions and variable names follow Python naming conventions.\r\n* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).\r\n* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.\r\n* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.\r\n* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.\r\n", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1745/timeline", "performed_via_github_app": null, "state_reason": null}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634/events", "html_url": "https://github.com/TheAlgorithms/Python/pull/1634", "id": 537914298, "node_id": "MDExOlB1bGxSZXF1ZXN0MzUzMTc1NjUy", "number": 1634, "title": "Bug Fixed in newton_raphson_method.py", "user": {"login": "rohitjoshi21", "id": 34398948, "node_id": "MDQ6VXNlcjM0Mzk4OTQ4", "avatar_url": "https://avatars.githubusercontent.com/u/34398948?v=4", "gravatar_id": "", "url": "https://api.github.com/users/rohitjoshi21", "html_url": "https://github.com/rohitjoshi21", "followers_url": "https://api.github.com/users/rohitjoshi21/followers", "following_url": "https://api.github.com/users/rohitjoshi21/following{/other_user}", "gists_url": "https://api.github.com/users/rohitjoshi21/gists{/gist_id}", "starred_url": "https://api.github.com/users/rohitjoshi21/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/rohitjoshi21/subscriptions", "organizations_url": "https://api.github.com/users/rohitjoshi21/orgs", "repos_url": "https://api.github.com/users/rohitjoshi21/repos", "events_url": "https://api.github.com/users/rohitjoshi21/events{/privacy}", "received_events_url": "https://api.github.com/users/rohitjoshi21/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 683905120, "node_id": "MDU6TGFiZWw2ODM5MDUxMjA=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/awaiting%20changes", "name": "awaiting changes", "color": "c9e5ff", "default": false, "description": "A maintainer has requested changes to the marked PR."}, {"id": 1473807336, "node_id": "MDU6TGFiZWwxNDczODA3MzM2", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/tests%20are%20failing", "name": "tests are failing", "color": "d93f0b", "default": false, "description": "Do not merge until tests pass"}], "state": "closed", "locked": false, "assignee": {"login": "rohitjoshi21", "id": 34398948, "node_id": "MDQ6VXNlcjM0Mzk4OTQ4", "avatar_url": "https://avatars.githubusercontent.com/u/34398948?v=4", "gravatar_id": "", "url": "https://api.github.com/users/rohitjoshi21", "html_url": "https://github.com/rohitjoshi21", "followers_url": "https://api.github.com/users/rohitjoshi21/followers", "following_url": "https://api.github.com/users/rohitjoshi21/following{/other_user}", "gists_url": "https://api.github.com/users/rohitjoshi21/gists{/gist_id}", "starred_url": "https://api.github.com/users/rohitjoshi21/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/rohitjoshi21/subscriptions", "organizations_url": "https://api.github.com/users/rohitjoshi21/orgs", "repos_url": "https://api.github.com/users/rohitjoshi21/repos", "events_url": "https://api.github.com/users/rohitjoshi21/events{/privacy}", "received_events_url": "https://api.github.com/users/rohitjoshi21/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "rohitjoshi21", "id": 34398948, "node_id": "MDQ6VXNlcjM0Mzk4OTQ4", "avatar_url": "https://avatars.githubusercontent.com/u/34398948?v=4", "gravatar_id": "", "url": "https://api.github.com/users/rohitjoshi21", "html_url": "https://github.com/rohitjoshi21", "followers_url": "https://api.github.com/users/rohitjoshi21/followers", "following_url": "https://api.github.com/users/rohitjoshi21/following{/other_user}", "gists_url": "https://api.github.com/users/rohitjoshi21/gists{/gist_id}", "starred_url": "https://api.github.com/users/rohitjoshi21/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/rohitjoshi21/subscriptions", "organizations_url": "https://api.github.com/users/rohitjoshi21/orgs", "repos_url": "https://api.github.com/users/rohitjoshi21/repos", "events_url": "https://api.github.com/users/rohitjoshi21/events{/privacy}", "received_events_url": "https://api.github.com/users/rohitjoshi21/received_events", "type": "User", "site_admin": false}], "milestone": null, "comments": 9, "created_at": "2019-12-14T13:29:54Z", "updated_at": "2019-12-15T07:27:08Z", "closed_at": "2019-12-15T07:27:08Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "draft": false, "pull_request": {"url": "https://api.github.com/repos/TheAlgorithms/Python/pulls/1634", "html_url": "https://github.com/TheAlgorithms/Python/pull/1634", "diff_url": "https://github.com/TheAlgorithms/Python/pull/1634.diff", "patch_url": "https://github.com/TheAlgorithms/Python/pull/1634.patch", "merged_at": "2019-12-15T07:27:08Z"}, "body": "Without using math function,function like sin and exp cannot be evaluated.\r\nAdding PRECISION variable explicitly helps user to play with precision.", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/1634/timeline", "performed_via_github_app": null, "state_reason": null}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301/events", "html_url": "https://github.com/TheAlgorithms/Python/pull/301", "id": 324728453, "node_id": "MDExOlB1bGxSZXF1ZXN0MTg5MjQ0ODM1", "number": 301, "title": "Create merge_sort_fastest.py", "user": {"login": "yesIamHasi", "id": 28681929, "node_id": "MDQ6VXNlcjI4NjgxOTI5", "avatar_url": "https://avatars.githubusercontent.com/u/28681929?v=4", "gravatar_id": "", "url": "https://api.github.com/users/yesIamHasi", "html_url": "https://github.com/yesIamHasi", "followers_url": "https://api.github.com/users/yesIamHasi/followers", "following_url": "https://api.github.com/users/yesIamHasi/following{/other_user}", "gists_url": "https://api.github.com/users/yesIamHasi/gists{/gist_id}", "starred_url": "https://api.github.com/users/yesIamHasi/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/yesIamHasi/subscriptions", "organizations_url": "https://api.github.com/users/yesIamHasi/orgs", "repos_url": "https://api.github.com/users/yesIamHasi/repos", "events_url": "https://api.github.com/users/yesIamHasi/events{/privacy}", "received_events_url": "https://api.github.com/users/yesIamHasi/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151092, "node_id": "MDU6TGFiZWw0MDkxNTEwOTI=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/duplicate", "name": "duplicate", "color": "cccccc", "default": true, "description": "This is a duplicate pull request from the same user"}, {"id": 409151093, "node_id": "MDU6TGFiZWw0MDkxNTEwOTM=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/enhancement", "name": "enhancement", "color": "696969", "default": true, "description": "This PR modified some existing files."}, {"id": 901901226, "node_id": "MDU6TGFiZWw5MDE5MDEyMjY=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/on%20hold", "name": "on hold", "color": "F0F8FF", "default": false, "description": ""}], "state": "closed", "locked": false, "assignee": {"login": "christianbender", "id": 23243382, "node_id": "MDQ6VXNlcjIzMjQzMzgy", "avatar_url": "https://avatars.githubusercontent.com/u/23243382?v=4", "gravatar_id": "", "url": "https://api.github.com/users/christianbender", "html_url": "https://github.com/christianbender", "followers_url": "https://api.github.com/users/christianbender/followers", "following_url": "https://api.github.com/users/christianbender/following{/other_user}", "gists_url": "https://api.github.com/users/christianbender/gists{/gist_id}", "starred_url": "https://api.github.com/users/christianbender/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/christianbender/subscriptions", "organizations_url": "https://api.github.com/users/christianbender/orgs", "repos_url": "https://api.github.com/users/christianbender/repos", "events_url": "https://api.github.com/users/christianbender/events{/privacy}", "received_events_url": "https://api.github.com/users/christianbender/received_events", "type": "User", "site_admin": false}, "assignees": [{"login": "christianbender", "id": 23243382, "node_id": "MDQ6VXNlcjIzMjQzMzgy", "avatar_url": "https://avatars.githubusercontent.com/u/23243382?v=4", "gravatar_id": "", "url": "https://api.github.com/users/christianbender", "html_url": "https://github.com/christianbender", "followers_url": "https://api.github.com/users/christianbender/followers", "following_url": "https://api.github.com/users/christianbender/following{/other_user}", "gists_url": "https://api.github.com/users/christianbender/gists{/gist_id}", "starred_url": "https://api.github.com/users/christianbender/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/christianbender/subscriptions", "organizations_url": "https://api.github.com/users/christianbender/orgs", "repos_url": "https://api.github.com/users/christianbender/repos", "events_url": "https://api.github.com/users/christianbender/events{/privacy}", "received_events_url": "https://api.github.com/users/christianbender/received_events", "type": "User", "site_admin": false}], "milestone": null, "comments": 3, "created_at": "2018-05-20T18:01:13Z", "updated_at": "2018-05-21T14:38:36Z", "closed_at": "2018-05-21T14:38:36Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "draft": false, "pull_request": {"url": "https://api.github.com/repos/TheAlgorithms/Python/pulls/301", "html_url": "https://github.com/TheAlgorithms/Python/pull/301", "diff_url": "https://github.com/TheAlgorithms/Python/pull/301.diff", "patch_url": "https://github.com/TheAlgorithms/Python/pull/301.patch", "merged_at": "2018-05-21T14:38:36Z"}, "body": "Python implementation of merge sort algorithm.\r\nTakes an average of 0.6 microseconds to sort a list of length 1000 items.\r\nBest Case Scenario : O(n)\r\nWorst Case Scenario : O(n)", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/301/timeline", "performed_via_github_app": null, "state_reason": null}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/291", "id": 314455357, "node_id": "MDU6SXNzdWUzMTQ0NTUzNTc=", "number": 291, "title": "ProjectEuler -- Problem 1 -- solv4.py -- Error", "user": {"login": "christianbender", "id": 23243382, "node_id": "MDQ6VXNlcjIzMjQzMzgy", "avatar_url": "https://avatars.githubusercontent.com/u/23243382?v=4", "gravatar_id": "", "url": "https://api.github.com/users/christianbender", "html_url": "https://github.com/christianbender", "followers_url": "https://api.github.com/users/christianbender/followers", "following_url": "https://api.github.com/users/christianbender/following{/other_user}", "gists_url": "https://api.github.com/users/christianbender/gists{/gist_id}", "starred_url": "https://api.github.com/users/christianbender/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/christianbender/subscriptions", "organizations_url": "https://api.github.com/users/christianbender/orgs", "repos_url": "https://api.github.com/users/christianbender/repos", "events_url": "https://api.github.com/users/christianbender/events{/privacy}", "received_events_url": "https://api.github.com/users/christianbender/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151094, "node_id": "MDU6TGFiZWw0MDkxNTEwOTQ=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/help%20wanted", "name": "help wanted", "color": "128A0C", "default": true, "description": null}, {"id": 902564907, "node_id": "MDU6TGFiZWw5MDI1NjQ5MDc=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/good%20first%20issue", "name": "good first issue", "color": "50d147", "default": true, "description": ""}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 1, "created_at": "2018-04-15T21:49:17Z", "updated_at": "2018-04-15T21:53:12Z", "closed_at": "2018-04-15T21:53:12Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "The correct answer is ```233168``` the program displays ```266333```.  See [here](https://github.com/TheAlgorithms/Python/blob/master/Project%20Euler/Problem%2001/sol4.py)  ", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/291/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/289", "id": 314453702, "node_id": "MDU6SXNzdWUzMTQ0NTM3MDI=", "number": 289, "title": "ProjectEuler -- Problem 1 -- solv2.py -- Error", "user": {"login": "christianbender", "id": 23243382, "node_id": "MDQ6VXNlcjIzMjQzMzgy", "avatar_url": "https://avatars.githubusercontent.com/u/23243382?v=4", "gravatar_id": "", "url": "https://api.github.com/users/christianbender", "html_url": "https://github.com/christianbender", "followers_url": "https://api.github.com/users/christianbender/followers", "following_url": "https://api.github.com/users/christianbender/following{/other_user}", "gists_url": "https://api.github.com/users/christianbender/gists{/gist_id}", "starred_url": "https://api.github.com/users/christianbender/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/christianbender/subscriptions", "organizations_url": "https://api.github.com/users/christianbender/orgs", "repos_url": "https://api.github.com/users/christianbender/repos", "events_url": "https://api.github.com/users/christianbender/events{/privacy}", "received_events_url": "https://api.github.com/users/christianbender/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151094, "node_id": "MDU6TGFiZWw0MDkxNTEwOTQ=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/help%20wanted", "name": "help wanted", "color": "128A0C", "default": true, "description": null}, {"id": 902564907, "node_id": "MDU6TGFiZWw5MDI1NjQ5MDc=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/good%20first%20issue", "name": "good first issue", "color": "50d147", "default": true, "description": ""}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 0, "created_at": "2018-04-15T21:27:04Z", "updated_at": "2018-04-16T19:22:01Z", "closed_at": "2018-04-16T19:22:01Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "For the Input ```1000``` I get ```233366.4```. The correct answer should be ```233168``` \r\nSee [file](https://github.com/TheAlgorithms/Python/blob/master/Project%20Euler/Problem%2001/sol2.py)", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/289/timeline", "performed_via_github_app": null, "state_reason": "completed"}, {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288", "repository_url": "https://api.github.com/repos/TheAlgorithms/Python", "labels_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288/labels{/name}", "comments_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288/comments", "events_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288/events", "html_url": "https://github.com/TheAlgorithms/Python/issues/288", "id": 314434745, "node_id": "MDU6SXNzdWUzMTQ0MzQ3NDU=", "number": 288, "title": "server.py : Error", "user": {"login": "christianbender", "id": 23243382, "node_id": "MDQ6VXNlcjIzMjQzMzgy", "avatar_url": "https://avatars.githubusercontent.com/u/23243382?v=4", "gravatar_id": "", "url": "https://api.github.com/users/christianbender", "html_url": "https://github.com/christianbender", "followers_url": "https://api.github.com/users/christianbender/followers", "following_url": "https://api.github.com/users/christianbender/following{/other_user}", "gists_url": "https://api.github.com/users/christianbender/gists{/gist_id}", "starred_url": "https://api.github.com/users/christianbender/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/christianbender/subscriptions", "organizations_url": "https://api.github.com/users/christianbender/orgs", "repos_url": "https://api.github.com/users/christianbender/repos", "events_url": "https://api.github.com/users/christianbender/events{/privacy}", "received_events_url": "https://api.github.com/users/christianbender/received_events", "type": "User", "site_admin": false}, "labels": [{"id": 409151091, "node_id": "MDU6TGFiZWw0MDkxNTEwOTE=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/bug", "name": "bug", "color": "ee0701", "default": true, "description": null}, {"id": 409151094, "node_id": "MDU6TGFiZWw0MDkxNTEwOTQ=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/help%20wanted", "name": "help wanted", "color": "128A0C", "default": true, "description": null}, {"id": 902564907, "node_id": "MDU6TGFiZWw5MDI1NjQ5MDc=", "url": "https://api.github.com/repos/TheAlgorithms/Python/labels/good%20first%20issue", "name": "good first issue", "color": "50d147", "default": true, "description": ""}], "state": "closed", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2018-04-15T17:27:48Z", "updated_at": "2018-04-16T11:41:20Z", "closed_at": "2018-04-16T05:00:51Z", "author_association": "CONTRIBUTOR", "active_lock_reason": null, "body": "See the file [here](https://github.com/TheAlgorithms/Python/blob/master/server.py)\r\n\r\n> Traceback (most recent call last):\r\n  File \"/home/christian/Programmierung/GitHub/Python/server.py\", line 29, in <module>\r\n    t.bind()\r\n  File \"/usr/lib/python2.7/socket.py\", line 228, in meth\r\n    return getattr(self._sock,name)(*args)\r\nTypeError: bind() takes exactly one argument (0 given)\r\n\r\nThe error above occures by python 2 and python 3. ", "reactions": {"url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288/reactions", "total_count": 0, "+1": 0, "-1": 0, "laugh": 0, "hooray": 0, "confused": 0, "heart": 0, "rocket": 0, "eyes": 0}, "timeline_url": "https://api.github.com/repos/TheAlgorithms/Python/issues/288/timeline", "performed_via_github_app": null, "state_reason": "completed"}]